### DISCUSSION TITLE
Custom Id Tags

### DISCUSSION THREAD

**Discussion by BrenMcM** [tinymce/tinymce#7949] [Ideas]
Hey guys. I know this has probably been asked, so apologies if I already missed such a discussion. But is there anyway at all to allow a custom unique Id to be added to tables generated by the tinymce By Default. The problem is we have a number of tables generated via code and others we want use tables generated via tinymce. In order for us to correctly restyle the tinymce tables for pdfs etc we need to be able to distinguish them all from the non-tinymce tables this would allow users with no html experience to create tables on our app. Currently the only way I am able to drop in some unique tags is either via the class selector (which is not really adequate for our setup) or via the extended-valid-attributes (which seems to destroy all other styling). 
Can I ask if there is a limitation for you guys on how this would be done?
Or is there something there already that I am missing?

**Comment by TheSpyder** [comment]
We don't have an easy way to generate a custom unique ID. We do have [`table_default_attributes`](https://www.tiny.cloud/docs/tinymce/6/table-options/#table_default_attributes) which can be used to add custom attributes, however as this value is an option it means the same value for each table. It would be difficult to dynamically update it to make the value unique for each table.

You could perhaps use this to set the attribute to a special value, and then I can think of two ways to replace that special value with a unique one:
* use selectors to locate tables with this custom attribute either in the [`BeforeGetContent`](https://www.tiny.cloud/docs/tinymce/6/events/#editor-core-events) event or at some other time that suits you
* monitor for [`mceInsertContent`](https://www.tiny.cloud/docs/tinymce/6/editor-command-identifiers/#miscellaneous-core-commands) events (which will always be a string) that are inserting tables
